app/kubernetes/views/applications/create/createApplication.html

```diff
@@ -47,1532 +53,1575 @@
               </div>
             </div>
             <!-- #endregion -->
-            <div class="col-sm-12 form-section-title">
-              Application
-            </div>
-            <!-- #region NAME FIELD -->
-            <div class="form-group">
-              <label for="application_name" class="col-sm-1 control-label text-left">Name</label>
-              <div class="col-sm-11">
-                <input
-                  type="text"
-                  class="form-control"
-                  name="application_name"
-                  ng-model="ctrl.formValues.Name"
-                  ng-change="ctrl.onChangeName()"
-                  placeholder="my-app"
-                  ng-pattern="/^[a-z]([-a-z0-9]*[a-z0-9])?$/"
-                  auto-focus
-                  required
-                  ng-disabled="ctrl.state.isEdit"
-                />
+
+            <!-- #region Git repository -->
+            <kubernetes-app-git-form
+              ng-if="ctrl.state.appType === ctrl.KubernetesDeploymentTypes.GIT"
+              show-config="ctrl.state.showConfig"
+              git-form-values="ctrl.gitFormValues"
+              is-edit="ctrl.state.isEdit"
+              on-change-ref="(ctrl.onChangeRef)"
+              on-change="(ctrl.onChange)"
+            ></kubernetes-app-git-form>
+            <!-- #endregion -->
+
+            <!-- #region web editor -->
+            <web-editor-form
+              ng-if="ctrl.state.appType === ctrl.KubernetesDeploymentTypes.CONTENT"
+              value="ctrl.stackFileContent"
+              yml="false"
+              identifier="kubernetes-deploy-editor"
+              placeholder="# Define or paste the content of your manifest file here"
+              on-change="(ctrl.onChangeFileContent)"
+            >
+              <editor-description>
+                <p>
+                  <i class="fa fa-info-circle blue-icon" aria-hidden="true" style="margin-right: 2px;"></i>
+                  This feature allows you to deploy any kind of Kubernetes resource in this environment (Deployment, Secret, ConfigMap...).
+                </p>
+                <p>
+                  You can get more information about Kubernetes file format in the
+                  <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/" target="_blank">official documentation</a>.
+                </p>
+              </editor-description>
+            </web-editor-form>
+            <!-- #endregion -->
+
+            <div ng-if="ctrl.state.appType === ctrl.KubernetesDeploymentTypes.APPLICATION_FORM">
```

Comment:
if we're already movingrefactoing this page, why not creat a component `KubeAppForm`?